### PR TITLE
Remove some unused modules

### PR DESCRIPTION
**Pull request by AlexKurek**
Resolves [JP-nnnn](https://jira.stsci.edu/browse/JP-nnnn)

Closes #

This PR addresses ...

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by emolter** [comment]
started [regression test run](https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1607/) here

File: jwst/fits_generator/main.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 # DAMAGE.
 
 # THIRD-PARTY
-from astropy.io import fits as pyfits
+# from astropy.io import fits as pyfits
```
Comment (emolter) [inline jwst/fits_generator/main.py:31]:
```suggestion
```

**Review by emolter — COMMENTED** [review]
These changes look good to me; indeed, none of these modules are used.  It seems to beg the question of why our ruff check didn't catch these earlier.  @zacharyburnett maybe that's a question for you?

**Comment by zacharyburnett** [comment]
> These changes look good to me, pending regtests; indeed, none of these modules are used. It seems to beg the question of why our ruff check didn't catch these earlier. @zacharyburnett maybe that's a question for you?

it looks like `fits_generator` and `associations` are excluded: https://github.com/spacetelescope/jwst/blob/8af184cef5fe4553ae704cd1932d494beca07272/pyproject.toml#L276-L285

I copied this config from the previous `flake8` config when moving it to `pyproject.toml`; I'm not sure why they were excluded in the first place

File: jwst/fits_generator/main.py (hunk 1 above)
Comment (AlexKurek) [reply jwst/fits_generator/main.py]: Done in https://github.com/spacetelescope/jwst/pull/8630/commits/aae54e248caeab4aa099e93b9dd31e318ee76d1f

**Comment by zacharyburnett** [comment]
I ran rules `F401` and `F841` on `jwst/fits_generator` and `jwst/associations` and it found quite a few: https://github.com/spacetelescope/jwst/pull/8651

**Review by melanieclarke — COMMENTED** [review]
These changes look good to me as well, and it does also sound like a good idea to remove those exclusions from the style checks.  Thanks for spotting that, @emolter, and taking care of it @zacharyburnett.

**Comment by emolter** [comment]
> looks like these exclusions were added in [9be1e27](https://github.com/spacetelescope/jwst/commit/9be1e2793ead8eebd270345d6db97012cfd7f155)

I think it was even earlier - that commit just split the exclusions across multiple lines.

Thanks for starting this discussion @AlexKurek - it sounds like we should go ahead and merge this one, and then look at https://github.com/spacetelescope/jwst/pull/8651 separately.  Does that sound ok to you @zacharyburnett ?

**Comment by zacharyburnett** [comment]
> > looks like these exclusions were added in [9be1e27](https://github.com/spacetelescope/jwst/commit/9be1e2793ead8eebd270345d6db97012cfd7f155)
> 
> I think it was even earlier - that commit just split the exclusions across multiple lines.
> 
> Thanks for starting this discussion @AlexKurek - it sounds like we should go ahead and merge this one assuming the regression tests don't turn up any surprises, and then look at #8651 separately. Does that sound ok to you @zacharyburnett ?

sounds good to me!

**Comment by zacharyburnett** [comment]
> These changes look good to me as well, and it does also sound like a good idea to remove those exclusions from the style checks. Thanks for spotting that, @emolter, and taking care of it @zacharyburnett.

made a separate PR to remove those exclusions https://github.com/spacetelescope/jwst/pull/8652

**Review by emolter — APPROVED** [review]
Regression tests are clean so I'm happy with this getting merged

**Pull request merged**